.github/workflows/e2e.yml

```diff
@@ -142,14 +142,22 @@ jobs:
       - name: Install ilab
         run: |
           export PATH="/home/runner/.local/bin:/usr/local/cuda/bin:$PATH"
-          python3 -m venv venv
+          python3.11 -m venv --upgrade-deps venv
           . venv/bin/activate
-          sed 's/\[.*\]//' requirements.txt > constraints.txt
-          python3 -m pip cache remove llama_cpp_python
-          CMAKE_ARGS="-DLLAMA_CUBLAS=on" python3 -m pip install --no-binary llama_cpp_python -c constraints.txt llama_cpp_python
-          # needed for --4-bit-quant option to ilab model train
-          python3 -m pip install bitsandbytes
-          python3 -m pip install .
+          nvidia-smi
+          python3.11 -m pip cache remove llama_cpp_python
+
+          CMAKE_ARGS="-DLLAMA_CUDA=on" python3.11 -m pip install .
+
+          # issue 1821: workaround for flash-attn
+          # install with Torch, packaging, and wheel present in the virtual env
+          python3.11 -m pip install packaging wheel
+          python3.11 -m pip install --no-build-isolation .[cuda]
```

Comment:
This install seems to result in instructlab reporting itself as version 0.0.0

https://github.com/instructlab/instructlab/actions/runs/10071643467/job/27842124384#step:16:970
https://github.com/instructlab/instructlab/actions/runs/10071643467/job/27842124384#step:17:21

Reply:
mmh, probably missing setuptools-scm. Let me check...